src/postgres/inventory/collection/PgCollectionKey.ts

```diff
@@ -134,10 +135,16 @@ class PgCollectionKey implements CollectionKey<PgClassType.Value, PgCollectionKe
    * @private
    */
   @memoizeMethod
-  private _getSelectLoader (client: Client): DataLoader<PgClassType.Value, PgClassType.Value | null> {
+  private _getSelectLoader (client: Client, resolveInfo: mixed, collectionGqlType: mixed): DataLoader<PgClassType.Value, PgClassType.Value | null> {
     return new DataLoader<PgClassType.Value, PgClassType.Value | null>(
-      async (keys: Array<PgClassType.Value>): Promise<Array<PgClassType.Value | null>> => {
+      async (keysAndStuff: mixed): Promise<Array<PgClassType.Value | null>> => {
```

Comment:
Yep, I think you’re right. We can remove `DataLoader` and have the same number of queries effectively. For the meantime, we can think about implementing a batch function on a field by field basis though.